File: 프로그래머스/lv0/120860. 직사각형 넓이 구하기/직사각형 넓이 구하기.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<vector<int>> dots)
{
    vector<int> xBig;
    vector<int> yBig;

    for (int i = 0; i < dots.size(); i++)
    {
        xBig.push_back(dots[i][0]);
        yBig.push_back(dots[i][1]);
    }

    sort(xBig.begin(), xBig.end());
    sort(yBig.begin(), yBig.end());


    int a = xBig[xBig.size()-1] - xBig[0];
    int b = yBig[yBig.size()-1] - yBig[0];

    return a * b;
}

File: 프로그래머스/0/120860. 직사각형 넓이 구하기/직사각형 넓이 구하기.java
class Solution {

    public int solution(int[][] dots) {
        int x = Math.max(Math.max(Math.max(dots[0][0], dots[1][0]), dots[2][0]), dots[3][0]);
        int x2 = Math.min(Math.min(Math.min(dots[0][0], dots[1][0]), dots[2][0]), dots[3][0]);
        int y = Math.max(Math.max(Math.max(dots[0][1], dots[1][1]), dots[2][1]), dots[3][1]);
        int y2 = Math.min(Math.min(Math.min(dots[0][1], dots[1][1]), dots[2][1]), dots[3][1]);
        return (x - x2) * (y - y2);
    }
}